### ISSUE TITLE
Duplicate identity, entities unresponsive

### ISSUE THREAD

**Issue by cuzzo333** [hashicorp/vault#11169]
**Describe the bug**
We're experiencing an odd issue with one of our applications authenticating to Vault in order to issue a token.  The application logs show a connection timeout to our Vault instance and Vault logs with timestamps matching these auth requests show 403 forbidden warnings.

Streaming logs via the CLI during these auth attempts results in tens of thousands of the following error (I've sanitized ID's and names):
[WARN]  identity: duplicate identity name: alias_name=<sanitized> mount_accessor=<sanitized>  entity_name=<sanitized>  action="delete one of the duplicate aliases"

Furthermore, when attempting to perform some further troubleshooting of the entity mentioned in the above log, we're seeing unresponsive behavior from the Entities menu in the web UI and the identity/entity API endpoint via the CLI.

**To Reproduce**
1. From the web UI attempting to access the Entities tab results in:
![image](https://user-images.githubusercontent.com/36944198/111891393-0b31e200-89b0-11eb-9e6d-aa96252762cb.png)
Attempting to troubleshoot the entity named in the log from the CLI with something such as the following command results in:
```
❯ vault read --format=json identity/entity/name/<sanitized>  | jq .data.aliases
Error reading identity/entity/name/<sanitized> : Error making API request.

URL: GET https://auth.nusutus.com/v1/identity/entity/name/<sanitized> 
Code: 503. Raw Message:

upstream connect error or disconnect/reset before headers. reset reason: connection termination
```

**Expected behavior**
We'd expect to be able to navigate to this area of the web UI or make API calls against this API endpoint.  Furthermore these issues seem related and to be the cause or at least symptoms of the authentication error we're seeing for the application in question.  Our other applications and services which tie into Vault are not experiencing any issues.

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.6.0
* Vault CLI Version (retrieve with `vault version`): 1.6.0
* Server Operating System/Architecture: Official Vault docker image running on Google Cloud Run

**Comment by HridoyRoy** [comment]
Hi @cuzzo333 , I'm not able to repro this. Can you please elaborate on the repro steps? 

From googling and looking through vault error return messages, it seems that this error is returned by something within Google Cloud Run, rather than an application error.
Thanks!

**Comment by sgmiller** [comment]
Closing as stale, but please reopen if you have new information.

**Comment by ncabatoff** [comment]
I think there are potentially a bunch of different phenomena grouped together here which are not necessarily related:
* logs about duplicate identities
* logs with 403 forbidden
* 503 errors returned to vault clients

My hunch is that there is no connection between these events.  The duplicate identities warning is just a warning and shouldn't lead to auth or connection failures as far as I know.  Why the 403 errors are occurring is unknown but it could be any number of things.

The 503 errors are the most alarming: these are generally only emitted when Vault is sealed or if a standby is handling a request and there is no leader for them to forward to (or they don't know who the leader is).  I suggest you focus your energies on investigating that problem first.

**Comment by cuzzo333** [comment]
@ncabatoff Thanks for the reply and what you said definitely makes sense.  My greatest concern at this time is the duplicate identities warning due to that tens of thousands of these warnings are generated in what seems like only several minutes.  I believe this is in turn causing a memory usage issue we're experiencing.  

We began seeing memory usage for our Vault instance spike a while back and gradually kept increasing and never decreasing, currently sitting at about 95% utilization of the 8GB of memory dedicated to the instance.  As we're running Vault on Google Cloud run, my assumption is that these logs are being written to memory since memory is acting as disk space.

This is where my troubleshooting began, trying to resolve the memory issue.  We can see the duplicate identity logs in both Google Cloud logging as well as receiving the output using the `vault monitor` command.  Seeing how many logs were being generated due to the duplicate identity, I began to try to delete the duplicate identity which is where I ran into the 503 errors using the API, the UI, etc.  These 503's are also given even if I just try to view/read the entityid which is being show in the logs.

So it seems I'm a it stuck.  If my assumptions are correct and the very high volume of logs being generated by the duplicate identity is filling up our memory, I need to be able to fix that issue but I appear to be unable to do so due to whatever is causing the 503 errors when I attempt to delete the duplicate identity.

**Comment by qk4l** [comment]
Today I've _renamed_ (created a new one with other name) JWT default role for our primary auth backend  and got `duplicate identity name` error.

```
Aug  4 14:00:02 vault.hostname vault[13044]: 2021-08-04T14:00:02.848+0300 [WARN]  identity: duplicate identity name: alias_name=client-hostname mount_accessor=auth_vault-plugin-auth-jwt_9350a7a9 entity_name=entity_ef109115 action="delete one of the duplicate aliases"
```

I checked that entity and found that it contains old role name. Looks like collision.

I also have 500 error when trying to list all entities but sure that it is a audit problem (we use syslog over UDP #11946 )

**Comment by hsimon-hashicorp** [comment]
Hi folks, this issue looks like it's had a bit of "scope creep" associated with it - can we get a little clarification as to what the issue is currently? Thanks!

**Comment by hsimon-hashicorp** [comment]
Hi! It's been a while since we've heard from you regarding this issue. I'm going to go ahead and close it now, but please feel free to re-open if you need and you can provide reproduction steps. In the meantime, our [Discuss forum](https://discuss.hashicorp.com/c/vault/30) is a great place to get help from subject matter experts and other community members, and is less of a "bug-reporting" venue. Thanks for your understanding!

**Comment by viniciusgarcia-hotmart** [comment]
Hey Guys, I'm facing this issue as well on kubernetes auth method. Since I'm using IRSA in my cluster I put the issuer at the kubernetes auth configuration the same of my cluster so I can use the AWS_WEB_IDENTITY_TOKEN_FILE token to authenticate against vault. Do we have some news about this error?

**Comment by cuzzo333** [comment]
> Hey Guys, I'm facing this issue as well on kubernetes auth method. Since I'm using IRSA in my cluster I put the issuer at the kubernetes auth configuration the same of my cluster so I can use the AWS_WEB_IDENTITY_TOKEN_FILE token to authenticate against vault. Do we have some news about this error?

My team has since moved on from this issue.  We never were able to solve it and expected some possible corruption in the storage backend we were using.  Due to a shift in technologies we've since migrated cloud providers and storage backends, previously running Vault using GCP Cloud Run with a GCS backend and now using AWS EC2 instances with EBS volumes as the backend.  We made this migration a few months ago and this issue has not been replicated since.  Wish I could help with a solution to this.
